src/lib/helpers/formatCurrency.ts

```diff
@@ -3,20 +3,26 @@ import { formatUnits } from "ethers/lib/utils.js"
 import { localeNumber } from "@/lib/helpers"
 
 export const formatCurrencyUnits = ({
-  abbreviate = false,
   amountWei = "0",
   decimals = 18,
+  maximumFractionDigits,
 }: {
-  abbreviate?: boolean
   amountWei?: string
   decimals?: number
-}) =>
-  abbreviate
-    ? localeNumber(Number(formatUnits(amountWei, decimals)), {
-        compact: true,
-        maximumFractionDigits: 2,
-      })
-    : formatUnits(amountWei, decimals)
+  maximumFractionDigits?: number
```

Comment:
In many cases we really need to show an abbreviated/truncated asset balance with more precision than 2 fraction digits allowed. This change makes it configurable for different use cases.